### PR TITLE
Ci 574 paginate apprenticeships on provider details page

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/Sfa.Das.ApprenticeshipInfoService.Core/Helpers/PaginationHelper.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            while (skip >= totalCount)
+            {
+                skip = skip - pageSize;
+            }
+
+            paginationDetails.LastPage = (totalCount + pageSize - 1) / pageSize;
+            paginationDetails.NumberOfRecordsToSkip = skip;
+            paginationDetails.Page = (skip / pageSize) + 1;
+            return paginationDetails;
+        }
```
Comment (froberg) [inline src/Sfa.Das.ApprenticeshipInfoService.Core/Helpers/PaginationHelper.cs]: The only thing that is used here is the pages to skip value. 
Would it be possible that this method only returns that value?

File: src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/ProvidersController.cs [hunk 2]
```diff
@@ -130,70 +121,37 @@ public void Head(long ukprn)
...
-                    Name = framework.PathwayName,
-                    Level = framework.Level,
-                    Type = ApprenticeshipTrainingType.Framework.ToString(),
-                    TrainingType = ApprenticeshipTrainingType.Framework,
-                    Identifier = framework.FrameworkId
-                })
-                .ToList();
-        }
-
-        private IEnumerable<ApprenticeshipTraining> GetActiveStandardsForUkprn(long ukprn)
```
Comment (froberg) [inline src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/ProvidersController.cs]: Why is this removed?

Comment (MarkFCain) [reply src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/ProvidersController.cs]: I moved the processing logic that used this from the controller to the repository, so it is no longer used in the controller

File: src/Sfa.Das.ApprenticeshipInfoService.Core/Helpers/PaginationHelper.cs (hunk 1 above)
Comment (MarkFCain) [reply src/Sfa.Das.ApprenticeshipInfoService.Core/Helpers/PaginationHelper.cs]: The story requires the maximum page number to be available in the front end - there is also front end work to do in das-search that will consume this api. In addition, the pagesize to use is stored as a value in the api rather than the front end, so there needed to be a mechanism available to inform the front end various details, like what the last page is, what page you are on, how many pages you records you've skipped, what the page size is, and so on.

I rolled these up into the paginationDetails class

File: src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/ProviderRepository.cs [hunk 3]
```diff
@@ -296,5 +305,63 @@ public IEnumerable<FrameworkProviderSearchResultsItem> GetProvidersByFrameworkId
...
+
+            var apprenticeships = new List<ApprenticeshipTraining>();
+            apprenticeships.AddRange(GetActiveStandardsForUkprn(ukprn));
+            apprenticeships.AddRange(GetActiveFrameworksForUkprn(ukprn));
+
+            var totalCount = apprenticeships.Count;
+
+            var pageSize = _applicationSettings.PageSizeApprenticeshipSummary;
+            var paginationDetails = _paginationHelper.GeneratePaginationDetails(page, pageSize, totalCount);
+            apprenticeshipTrainingSummary.PaginationDetails = paginationDetails;
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/ProviderRepository.cs]: No, because this information is needed in the front end (in das-search)
